keyboards/sofle_choc/config.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2022 Brian Low
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+// Key matrix size
+// Rows are doubled-up
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 6
+
+// Wiring of each half
+#define MATRIX_ROW_PINS { C6, D7, E6, B4, B5 }
+#define MATRIX_COL_PINS { F6, F7, B1, B3, B2, B6 }
+#define DIODE_DIRECTION COL2ROW
+
+#define DEBOUNCE 5
+
+// Encoder support
+#define ENCODERS_PAD_A { F5 }
+#define ENCODERS_PAD_B { F4 }
+#define ENCODERS_PAD_A_RIGHT { F4 }
+#define ENCODERS_PAD_B_RIGHT { F5 }
+
+// Communication between sides
+#define SOFT_SERIAL_PIN D2
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_DI_PIN D3
+#define RGBLED_NUM 29
+#define DRIVER_LED_TOTAL 58
```

Comment:
```suggestion
#define RGB_MATRIX_LED_COUNT 58
```

Reply:
@waffle87  I couldn't find `RGB_MATRIX_LED_COUNT` in the docs and I get a compile error with the suggested change:
```
matrix/rgb_matrix.h:23,
                 from quantum/quantum.h:36,
                 from keyboards/sofle_choc/sofle_choc.h:19,
                 from keyboards/sofle_choc/sofle_choc.c:17:
quantum/rgb_matrix/rgb_matrix_types.h:81:23: error: 'DRIVER_LED_TOTAL' undeclared here (not in a function)
   81 |     led_point_t point[DRIVER_LED_TOTAL];
```

Reply:
Preemptively suggesting this as it will be the default on `master` branch when this cycle's [breaking changes](https://docs.qmk.fm/#/breaking_changes) are merged — within the next day or two. Sorry should've elaborated a bit.

Reply:
Thanks @waffle87. I'll update this PR when the breaking changes are merged to keep this branch passing CI.

Reply:
Updated to `RGB_MATRIX_LED_COUNT`, thanks for the heads up